### PR TITLE
Merge release/2.1 infrastructure changes into master

### PR DESCRIPTION
**Pull request by dagood**
The 2.1.2 release saw some changes in `release/2.1` infra to support VSTS builds. This merge takes those changes into `master` for the future. I made some edits to the merge commit to fix conflicts and remove any submodule upgrades related to the 2.1.2 release.

**Don't squash this PR.** Squashing https://github.com/dotnet/source-build/pull/577 (an earlier 2.1 => master merge) made some commits it included to show up in this PR's commit list, too.

### CODE REVIEW

File: repos/nuget-client.proj [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
   <ItemGroup>
     <EnvironmentVariables Include="MS_PFX_PATH=$(NuGetKeyFilePath)" />
+    <EnvironmentVariables Include="NUGET_PFX_PATH=$(NuGetKeyFilePath)" />
```
Comment (dagood) [inline repos/nuget-client.proj]: I wasn't sure about this when I first saw it. I see now it comes from https://github.com/dotnet/source-build/pull/598, so we probably do need it. @crummel

File: repos/dir.props [hunk 2]
```diff
@@ -53,6 +53,7 @@
     <EnvironmentVariables Include="DotNetCoreSdkDir=$(DotNetCliToolDir)" />
     <EnvironmentVariables Include="DotNetBuildToolsDir=$(ProjectDir)Tools/" />
     <EnvironmentVariables Include="StabilizePackageVersion=$(UseStableVersions)" Condition="'$(UseStableVersions)' != ''" />
+    <EnvironmentVariables Include="PB_IsStable=$(UseStableVersions)" Condition="'$(UseStableVersions)' != ''" />
```
Comment (dagood) [inline repos/dir.props]: This is from https://github.com/dotnet/source-build/commit/cdf7249ccb14bf1b1da499f0c283b78129556036, we do need it for proper versions.

File: netci.groovy [hunk 3]
```diff
@@ -44,7 +44,7 @@ def addBuildStepsAndSetMachineAffinity(def job, String os, String configuration)
         shell("git submodule update --init --recursive");
         shell("./build.sh /p:Configuration=${configuration} ${loggingOptions}");
         smokeTestExcludes = "";
-        if (os == "OSX10.12") {
+        if (os == "Fedora24" || os == "OSX10.12") {
```
Comment (dseefeld) [inline netci.groovy]: We probably don't need this change in master.  Master is updated to build on Fedora28, now.

Comment (dagood) [reply netci.groovy]: Good call, removed.

File: repos/nuget-client.proj (hunk 1 above)
Comment (crummel) [reply repos/nuget-client.proj]: Yeah, there's now two different variables used for the PFX path, we will attempt to delay-sign and fail without this.  I can look at whether we can disable signing entirely, that seems like the better option.

Comment (dagood) [reply repos/nuget-client.proj]: Cool, filed https://github.com/dotnet/source-build/issues/647.

**Pull request merged**